pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -144,7 +152,7 @@ func (l *ListOptionIndexer) afterUpsert(key string, obj any, tx db.TXClient) err
 			logrus.Errorf("cannot index object of type [%s] with key [%s] for indexer [%s]: %v", l.GetType().String(), key, l.GetName(), err)
 			cErr := tx.Cancel()
 			if cErr != nil {
-				return errors.Wrap(err, cErr.Error())
+				return fmt.Errorf("could not cancel transaction: %s while recovering from error: %w", cErr.Error(), err)
 			}
```

Comment:
The use of `%s` will trigger the call to `cErr.Error()` so you don't need to do it explicitly.

> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

Reply:
Right, thanks for the observation. Unfortunately commits in this PR were merged together as part of #84, so I am starting [a separate PR](https://github.com/rancher/lasso/pull/90) to address this feedback